core/policy/policy_repository.go

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package policy
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/mysteriumnetwork/node/market"
+	"github.com/mysteriumnetwork/node/requests"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+)
+
+// PolicyRepository represents async policy fetcher from TrustOracle
+type PolicyRepository struct {
+	client *requests.HTTPClient
+
+	policyURL   string
+	policyLock  sync.Mutex
+	policyRules map[market.AccessPolicy]market.AccessPolicyRuleSet
+
+	fetchInterval time.Duration
+	fetchShutdown chan bool
+}
+
+// NewPolicyRepository create instance of policy repository
+func NewPolicyRepository(client *requests.HTTPClient, policyURL string, interval time.Duration) *PolicyRepository {
+	return &PolicyRepository{
+		client:        client,
+		policyURL:     policyURL,
+		policyRules:   make(map[market.AccessPolicy]market.AccessPolicyRuleSet),
+		fetchInterval: interval,
+	}
+}
+
+// Start begins fetching proposals to repository
+func (pr *PolicyRepository) Start() {
+	pr.fetchShutdown = make(chan bool, 1)
```

Comment:
1. Why do you need buffered channel here?
2. Stop is usually done simpler with empty struct channel like `chan struct`, you will find it in 100 places in node repo. When just send `fetchShutdown <- struct{}` to stop. If you use bool it is confusing. For example can I send only true, or false is also acceptable.

Reply:
1. Why’d  not? This channel wait just for single stop message + don't want to block sending received isn't ready
2. Don't see a difference, sending as minimal structure as possible. Changed to s`truct{}`

Reply:
1. This stop is only called if node is shutting down. In such case there is no point to Stop in async fashion.
2. As minimal as possible is empty struct.